src/dmap_common.c

```diff
@@ -355,7 +355,7 @@ dmap_send_error(struct evhttp_request *req, const char *container, const char *e
     {
       DPRINTF(E_LOG, L_DMAP, "Could not allocate evbuffer for DMAP error\n");
 
-      evhttp_send_error(req, HTTP_SERVUNAVAIL, "Internal Server Error");
+      httpd_send_error(req, HTTP_SERVUNAVAIL, "Internal Server Error");
```

Comment:
Regarding adding the headers at several places in the code how would you implement that?

Should I add a new function like httpd_add_cors_headers(struct evkeyvalq *headers) and call it at several places or better to have a function like httpd_send_reply_internal which is more or less a wrapper around evhttp_send_reply like in my httpd_send_error and replace calls to evhttp_send_reply with this new function?